cnudie/iter.py

```diff
@@ -105,9 +107,12 @@ def iter(
                             dependencies; -1 will resolve w/o recursion limit, 0 will not resolve
                             component dependencies
     @param prune_unique: if true, redundant component-versions will only be traversed once
-    @node_filter:        use to filter emitted nodes (see Filter for predefined filters)
+    @param node_filter:  use to filter emitted nodes (see Filter for predefined filters)
     @param ocm_repo:     optional OCM Repository to be used to override in the lookup
     @param ctx_repo:     deprecated, use `ocm_repo` instead
+    @param component_filter: use to exclude components (and their references) from the iterator
```

Comment:
it would be nice to state that returning `True` means component should be filtered out